### PR TITLE
Update mtrand.pyx

### PR DESCRIPTION
**Pull request by orena1**
I know int is between 0 and 4294967295, but I think many people that do not know that will benefit from this comment.

### CODE REVIEW

File: numpy/random/mtrand/mtrand.pyx [hunk 1]
```diff
@@ -892,7 +892,7 @@ cdef class RandomState:
     ----------
     seed : {None, int, array_like}, optional
         Random seed initializing the pseudo-random number generator.
-        Can be an integer, an array (or other sequence) of integers of
+        Can be an integer (between 0 and 4294967295), an array (or other sequence) of integers of
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Get rid of the parenthesis.

**Comment by charris** [comment]
Good, but there is more ;) Squash the commits and rewrite the commit message to conform with doc/source/dev/gitwash/development_workflow.rst, put more explanation in the commit message and also add `[ci skip]`  as documentation is not tested. You can do that with `git rebase -i HEAD^^` followed by a force push.

**Comment by jaimefrio** [comment]
Throwing a magic number like that without context in the docs seems awkward to me. I would at least say somewhere that it is `2**32 - 1`, either explicitly or by noting that it is a 32 bit unsigned integer.

**Comment by njsmith** [comment]
Is the seed in fact 32 bits?

**Comment by orena1** [comment]
@njsmith yes it is, there is even a check if you try a bigger number:
```python
In [1]: rnd = random.RandomState(2**32)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-06a1db55777c> in <module>()
----> 1 rnd = random.RandomState(2**32)

mtrand.pyx in mtrand.RandomState.__init__ (numpy/random/mtrand/mtrand.c:8275)()

mtrand.pyx in mtrand.RandomState.seed (numpy/random/mtrand/mtrand.c:8670)()

ValueError: Seed must be between 0 and 4294967295

```

@jaimefrio   I think you are right, I will change the code accordingly
@charris I will learn how to do it, and will try again..

File: numpy/random/mtrand/mtrand.pyx [hunk 2]
```diff
@@ -892,7 +892,7 @@ cdef class RandomState:
     ----------
     seed : {None, int, array_like}, optional
         Random seed initializing the pseudo-random number generator.
-        Can be an integer, an array (or other sequence) of integers of
+        Can be an integer between 0 and 4294967295, an array (or other sequence) of integers of
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Keep the line length under 75 characters.

**Comment by charris** [comment]
Updated in #7359, so closing this. Thanks @orena1 .

**Pull request closed without merging**